src/librustdoc/clean/auto_trait.rs

```diff
@@ -354,9 +354,7 @@ where
         ty_to_bounds
             .into_iter()
             .flat_map(|(ty, mut bounds)| {
-                if let Some(data) = ty_to_fn.get(&ty) {
-                    let (poly_trait, output) =
-                        (data.0.as_ref().unwrap().clone(), data.1.as_ref().cloned().map(Box::new));
+                if let Some((Some(ref poly_trait), ref output)) = ty_to_fn.get(&ty) {
```

Comment:
I'm a bit concerned about this `poly_trait` change. Why would it ever be `None`? Shouldn't we always have access to the HRTB for the `Fn`?